command/server.go

```diff
@@ -79,6 +79,8 @@ type ServerCommand struct {
 	LogicalBackends    map[string]logical.Factory
 	PhysicalBackends   map[string]physical.Factory
 
+	PhysicalServiceDiscoverers map[string]physical.ServiceDiscoveryFactory
```

Comment:
I understand the reasoning behind the naming due to it being in the `physical` package, but can we call this `ServiceDiscoverers` instead (same with `physicalServiceDiscoverers`)?

Reply:
I moved ServiceDiscovery into its own package